src/thread_pool/thread_pool.ml

```diff
@@ -0,0 +1,75 @@
+open Stdune
```

Comment:
It would be nice to provide a little high-level specification of how the thread pool works/what guarantees it provides.